code/modules/projectiles/magazines/flamer.dm

```diff
@@ -90,7 +90,8 @@
 		to_chat(user, SPAN_WARNING("You can't mix fuel mixtures!"))
 		return
 
-	if(!to_add.intensityfire && to_add.id != "stablefoam")
+	var/is_smoke_tank = istype(src, /obj/item/ammo_magazine/flamer_tank/smoke)
+	if(!to_add.intensityfire && to_add.id != "stablefoam" && !is_smoke_tank)
```

Comment:
```suggestion
	if(!to_add.intensityfire && to_add.id != "stablefoam" && !istype(src, /obj/item/ammo_magazine/flamer_tank/smoke))
```
Its not as clean but via short-circuiting it reduces how much type checking is done is most scenarios.